client/infinispan-key-value-store-hotrod/src/test/java/org/infinispan/api/client/impl/SearchKeyValueStoreTest.java

```diff
@@ -0,0 +1,217 @@
+            .map(kv -> kv.value()).map(o -> (String) o[0])
+            .collect(Collectors.toList());
+
+      assertTrue(firstNames.contains(OIHANA.firstName));
+      assertTrue(firstNames.contains(DANIELA.firstName));
+      assertTrue(firstNames.contains(UNAI.firstName));
+      assertTrue(firstNames.contains(ELAIA.firstName));
+      assertTrue(firstNames.contains(MIREN.firstName));
+      assertTrue(firstNames.contains(EDOIA.firstName));
+   }
+
+   @Test
+   public void continuous_query_over() {
+      TestSubscriber<KeyValueEntry<String, Person>> personTestSubscriber = new TestSubscriber<>();
+      QueryRequest queryRequest = ContinuousQueryRequestBuilder.query("FROM org.infinispan.Person p")
+            .created().build();
+
+      Publisher<KeyValueEntry<String, Person>> continuously = store.findContinuously(queryRequest);
+
+      continuously.subscribe(personTestSubscriber);
+
+      personTestSubscriber.assertValueCount(6);
+   }
+
+   @Test
+   public void continuous_query_search_only_created_entries() {
+      TestSubscriber<KeyValueEntry<String, Person>> personTestSubscriber = new TestSubscriber<>();
+      QueryRequest queryRequest = ContinuousQueryRequestBuilder.query("FROM org.infinispan.Person p")
+            .created().build();
+
+      Publisher<KeyValueEntry<String, Person>> continuously = store.findContinuously(queryRequest);
+      continuously.subscribe(personTestSubscriber);
+
+
+      personTestSubscriber.assertSubscribed();
+
+      updateOneValue();
+
+      personTestSubscriber.assertValueCount(6);
+   }
+
+   @Test
+   public void continuous_query_search_updated_entries() throws Exception {
+      TestSubscriber<KeyValueEntry<String, Person>> personTestSubscriber = new TestSubscriber<>();
+      QueryRequest queryRequest = ContinuousQueryRequestBuilder.query("FROM org.infinispan.Person p")
+            .updated().build();
+
+      Publisher<KeyValueEntry<String, Person>> continuously = store.findContinuously(queryRequest);
+      continuously.subscribe(personTestSubscriber);
+
+      personTestSubscriber.assertSubscribed();
+      personTestSubscriber.assertValueCount(0);
+
+      updateOneValue();
+
+      personTestSubscriber.await(1, TimeUnit.SECONDS);
+      await(store.delete(OIHANA.id));
+      personTestSubscriber.assertValueCount(1);
+   }
+
+   @Test
+   public void continuous_query_search_deleted_entries() throws Exception {
+      TestSubscriber<KeyValueEntry<String, Person>> personTestSubscriber = new TestSubscriber<>();
+      QueryRequest queryRequest = ContinuousQueryRequestBuilder.query("FROM org.infinispan.Person p")
+            .deleted().build();
+
+      Publisher<KeyValueEntry<String, Person>> continuously = store.findContinuously(queryRequest);
+      continuously.subscribe(personTestSubscriber);
+
+      personTestSubscriber.assertSubscribed();
+      personTestSubscriber.assertValueCount(0);
+
+      await(store.clear());
+      personTestSubscriber.await(1, TimeUnit.SECONDS);
```

Comment:
this looks like a source of random failures to me...

Reply:
yes. I will fix these tests in the next PR